### DISCUSSION TITLE
file download not working

### DISCUSSION THREAD

**Discussion by jiecerbo** [wintercms/winter#977] [Q&A] [answered]
I can't seem to get the download link to work. when i click on the button to download the file, i only get a javascript alert with weird characters in the body.

In my ajax handler i'm using Response::download(pathtofile) and in my markup i have this:

`<button data-request="onExport" data-request-download data-browser-target="_blank">Download File </button>`

**Comment by LukeTowers** [comment, marked as answer]
AJAX handlers need to return a redirect to a controller that returns the download AFAIK, direct download over AJAX isn't natively supported by browsers I believe; at least the last time I checked it wasn't. Also where are you getting the `data-request-download` attribute from?
